File: javascript/ipk.hashmap.ransomNote.js
function checkMagazine(magazine, note) {
  var m = {};
  var n = {};
  for (var i = 0; i < magazine.length; i ++) {
    if (m[magazine[i]]) {
      m[magazine[i]] ++;
    } else {
      m[magazine[i]] = 1;
    }
  }
  for (var i = 0; i < note.length; i ++) {
    if (n[note[i]]) {
      n[note[i]] ++;
    } else {
      n[note[i]] = 1;
    }
  }
  for (var k in n) {
    if (!m[k]) return 'No';
    if (n[k] > m[k]) return 'No';
  }
  
  return 'Yes';
}


exports.checkMagazine = checkMagazine;

File: python/ipk.hashmap.ransomNote.py
def checkMagazine(magazine, note):
    m = dict()
    n = dict()
    for item in magazine:
        if item in m:
            m[item] = m[item] + 1
        else:
            m[item] = 1
    for item in note:
        if item in n:
            n[item] = n[item] + 1
        else:
            n[item] = 1
    for item in n:
        if item not in m:
            return "No"
        if n[item] > m[item]:
            return "No"
    return "Yes"
